compiler/qsc_qasm/src/semantic/lowerer.rs

```diff
@@ -655,7 +721,45 @@ impl Lowerer {
         }
     }
 
-    fn lower_lit_expr(&mut self, expr: &syntax::Lit) -> semantic::Expr {
+    fn check_lit_array_type(&mut self, expr: &syntax::Lit, ty: &Type) {
```

Comment:
Perhaps making this name specific to sizes would be good since it doesn't check the types just the structure.